### ISSUE TITLE
DFP [7.x-2.x] Maintainer

### ISSUE THREAD

**Issue by scottalan** [heshanlk/prebidjs#1]
I just came across your module and I'd like to help out. I maintain the [7.x-2.x branch](https://cgit.drupalcode.org/dfp/log/?h=7.x-2.x) of the DFP module. I noticed someone (maybe someone you know) grabbed the [same namespace](https://www.drupal.org/project/prebidjs). Is this where you are keeping development of this module or do you have a project page on D.O. as well?

I like the idea of using ctools exports to store data about each of the bidders. I will probably have to update a few things to get it to work for our client so I might be creating a few PRs. I had to do something similar for DFP which is why there is a 2.x branch.

**Comment by heshanlk** [comment]
Yes @scottalan I saw that but have no idea who that is so I opened this https://www.drupal.org/project/prebidjs/issues/2934453.

Feel free to fork and send me PRs or take full control 👍

**Comment by scottalan** [comment]
@heshanlk Thanks for responding. I'll see if I can't get some help moving that issue along.

**Comment by heshanlk** [comment]
That'd be great!. Let me know if I can help you with anything.

**Comment by scottalan** [comment]
@heshanlk, from what I can gather I'm seeing that there is a legacy version and I'm assuming this is anything **< 1.x.x**? It looks like even prebid.org is behind unless it is actually using the "latest" and they just haven't updated the option.

![ss-from-scott 2018-06-21 at 6 32 52 pm](https://user-images.githubusercontent.com/247016/41750546-3760b7e2-7582-11e8-850d-604b56b8f32e.png)

I wasn't sure what the versioning requirements were for the `master` branch but I'm required to use the "latest" (1.13.0) for the current project I'm working on. I can update bits and pieces here and there checking for the `$library['version']`, but I feel this could get out of control at some point.

Maybe `master` could become a `7.x-1.x-dev` on D.O. and maybe what I'm working on could be a `7.x-2.x-dev` branch  that requires the newer version?

Let me know what you think.

**Comment by heshanlk** [comment]
@scottalan the idea of using Libraries API was to remove that kind of version dependencies, don't you think using just the version you want in sites/all/libraries would do it?.

I can't remember how much the syntax is different for those two versions so feel free to brank off if that makes sense.

**Comment by scottalan** [comment]
I ended just adding a submodule (prebid) to the 7.x-2.x branch of DFP. The fact that there is no standardization to the bidder parameters is insane IMO. I opted to go with a plugin system via ctools that allows a module to create individual `bidder` plugins the handle unique use cases for each of the bidders.

For example:

One of the bidders we are using is `index exchange` IX. 
The following are the parameters it requires:
```
'params' => array(
    'siteId' => '',
    'size' => array(),
  ),
```
I had to define a different `siteId` for each of the ad slots. The plugin system allowed me to do this for this particular bidder.

Here's the patch we are testing now https://www.drupal.org/project/dfp/issues/2980845#comment-12730946
